### PR TITLE
RHIROS-1068 Store all recommendation sets

### PR DESCRIPTION
**Pull request by saltgen**
Changes to keep all recommendations for future analysis.

### CODE REVIEW

File: internal/model/recommendation_set.go [hunk 1]
```diff
@@ -96,12 +92,3 @@ func (r *RecommendationSet) CreateRecommendationSet() error {
...
 
-func DeleteStaleRecommendationSet(workload_id uint, containers []string) error {
-	db := database.GetDB()
-	result := db.Where("workload_id = ? AND container_name NOT IN ?", workload_id, containers).Delete(&RecommendationSet{})
-	if result.Error != nil {
-		return result.Error
-	}
-
-	return nil
-}
```
Comment (patilsuraj767) [inline internal/model/recommendation_set.go:107]: Why we are removing this?

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]
We should not completely remove the unique constraint. But modify it as below. 
```
ALTER TABLE recommendation_sets
ADD CONSTRAINT UQ_Recommendation UNIQUE (workload_id, container_name, monitoring_end_time);
```

File: internal/model/recommendation_set.go [hunk 2]
```diff
@@ -84,10 +83,7 @@ func (r *RecommendationSet) GetRecommendationSetByID(orgID string, recommendatio
 
 func (r *RecommendationSet) CreateRecommendationSet() error {
 	db := database.GetDB()
-	result := db.Clauses(clause.OnConflict{
-		Columns:   []clause.Column{{Name: "workload_id"}, {Name: "container_name"}},
-		DoUpdates: clause.AssignmentColumns([]string{"monitoring_start_time", "monitoring_end_time", "recommendations", "updated_at"}),
-	}).Create(r)
+	result := db.Create(r)
```
[the remark this answers is not in the document]
Comment (saltgen) [reply internal/model/recommendation_set.go:86]: I had this question regarding duplication for the Updates meet, thanks will modify and push

[the remark this answers is not in the document]
Comment (saltgen) [reply internal/model/recommendation_set.go:86]: `DoUpdates: clause.AssignmentColumns([]string{"monitoring_start_time", "recommendations", "updated_at"})`
Should we DoNothing instead?

File: internal/model/recommendation_set.go (hunk 1 above)
Comment (saltgen) [reply internal/model/recommendation_set.go:107]: :thinking: , would it be okay to get rid of recommendations for non-existing containers from a historical data pov?

File: internal/model/recommendation_set.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply internal/model/recommendation_set.go:86]: I think it is good to always update the recommendation set to the latest one which is coming from kruize.

File: internal/model/recommendation_set.go (hunk 1 above)
Comment (patilsuraj767) [reply internal/model/recommendation_set.go:107]: From historical data pov we should keep it. I was just think from user perspective that it should not happen we are showing some recommendation but actually that container does not exist in customer environment. 
Anyway I think we can remove this method for now, we will take care of stale data in further iterations.

File: internal/model/recommendation_set.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (saltgen) [reply internal/model/recommendation_set.go]: Changes pushed

**Review by patilsuraj767 — APPROVED** [review]
LGTM :+1: Thanks @saltgen

**Pull request merged**